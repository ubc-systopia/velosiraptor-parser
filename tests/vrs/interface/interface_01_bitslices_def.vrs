segment MyUnit {
  interface(base: foo) {
    // register with a simple slice
    reg foo [ 8 ] {
      Layout {
        0 .. 2 foo
      }
    },
    // double definition of the same register, should parse fine
    reg foo [ 8 ] {
      Layout {
        0 .. 2 foo,
        2 .. 4 bar,
      }
    },
    // memory location with three slices
    mem bar [ base, 0, 8 ] {
      Layout {
        0 .. 2 foo,
        // invalid slice range, should parse fine
        7 .. 4 bar,
        // double definition of the slice, should parse fine
        7 .. 4 bar,
      }
    },
    mmio baz [ base, 8, 8 ] {
      Layout {
        // exceeding ranges, should parse fine
        0 .. 256 foo,
        1025 .. 4 bar,
      },
      // two layout definitions are OK for parsing
      Layout {
        0 .. 2 foo
      }
    },
  }
}
