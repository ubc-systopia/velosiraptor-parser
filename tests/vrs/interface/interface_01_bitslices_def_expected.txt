VelosiParseTree(tests/vrs/interface/interface_01_bitslices_def.vrs)
---------------------------------------------
segment MyUnit {
  interface(base: foo) {
    reg foo [ 8 ] {
      Layout {
        0..2 foo,
      },
    },
    reg foo [ 8 ] {
      Layout {
        0..2 foo,
        2..4 bar,
      },
    },
    mem bar [ base, 0, 8 ] {
      Layout {
        0..2 foo,
        7..4 bar,
        7..4 bar,
      },
    },
    mmio baz [ base, 8, 8 ] {
      Layout {
        0..256 foo,
        1025..4 bar,
      },
      Layout {
        0..2 foo,
      },
    },
  }
}


---------------------------------------------
